srk/src/polyhedronLatticeTiling.ml

```diff
@@ -0,0 +1,1486 @@
+    let compose = LocalAbstraction.compose in
+    cubify
+    |> compose (PltConvexHull.by_polyreccone_and_lplh ~man ~epsilon ~max_dim_in_target)
+
+  let by_lplh ?(man=Polka.manager_alloc_loose()) ~epsilon srk terms symbols =
+    let (cubify, _) = Plt.cubify srk terms symbols in
+    let max_dim_in_target = Array.length terms - 1 in
+    let compose = LocalAbstraction.compose in
+    cubify
+    |> compose (PltConvexHull.local_project_local_hull ~man ~epsilon ~max_dim_in_target)
+
+  let by_polyreccone ?(man=Polka.manager_alloc_loose()) srk terms symbols =
+    let (cubify, _) = Plt.cubify srk terms symbols in
+    let max_dim_in_target = Array.length terms - 1 in
+    let compose = LocalAbstraction.compose in
+    cubify
+    |> compose (PltConvexHull.local_project_polyreccone ~man ~max_dim_in_target)
+
+end
+
+module LocalGlobal: sig
+
+  val lift:
+    man:DD.closed Apron.Manager.t ->
+    ('a formula, 'a Interpretation.interpretation, DD.closed DD.t, int -> QQ.t) LocalAbstraction.t ->
+    'a Abstract.Solver.t -> ?bottom: DD.closed DD.t -> ('a arith_term) Array.t -> DD.closed DD.t
+
+end = struct
+
+  let print_model srk terms interp =
+    let result =
+      Array.init (Array.length terms)
+        (fun i ->
+          (terms.(i), Interpretation.evaluate_term interp terms.(i)))
+    in
+    logf ~level:`debug "model: @[%a@]@;"
+      (Format.pp_print_list
+         ~pp_sep:(fun fmt () -> Format.fprintf fmt "; ")
+         (fun fmt (t, value) ->
+           Format.fprintf fmt "(%a, %a)"
+             (Syntax.ArithTerm.pp srk) t
+             QQ.pp value))
+      (Array.to_list result)
+
+  let lift ~man local_abs solver ?bottom terms =
+    let target_dim = Array.length terms in
+    let bottom =
+      match bottom with
+      | None -> P.dd_of ~man target_dim P.bottom
+      | Some bot -> bot
+    in
+    let top = P.dd_of ~man target_dim P.top in
+    let srk = Abstract.Solver.get_context solver in
+    let phi = Abstract.Solver.get_formula solver in
+    let counter = ref 0 in
+    let models = ref [] in
+    let show m =
+      let symbols =
+        Syntax.Symbol.Set.elements (Syntax.symbols phi)
+        |> List.map (Syntax.mk_const srk) |> Array.of_list
+      in
+      print_model srk symbols m
+    in
+    let of_model m =
+      match m with
+      | `LIRR _ -> invalid_arg "LIRR not supported"
+      | `LIRA m ->
+         let () = show m in
+         models := m :: !models;
+         counter := !counter + 1;
+
+         logf ~level:`debug "Abstraction loop iteration: %d" !counter;
+         let result = local_abs.LocalAbstraction.abstract m phi in
+         logf ~level:`debug "Abstraction loop iteration %d done" !counter;
+         result
+    in
+    let domain =
+      Abstract.
+      { join = DD.join
+      ; of_model
+      ; formula_of = formula_of_dd srk (fun i -> terms.(i))
+      ; top
+      ; bottom
+      }
+    in
+    Abstract.Solver.abstract solver domain
+
+end
+
+type lira_abstraction =
```

Comment:
Why  dowe want to reify algorithm selection into a datatype?  (I.e., why not ask the client to provide the "ConvexHull.by_polyreccone" as the "how" instead of "LiraCCH PolyReccone")
Some of these configurations are for evaluation purposes only, and will eventually be removed.

Reply:
Yeah. I will resolve some of the other comments first, then fork to have an evaluation branch that's for the experiments and submission. The one to be pulled into flint will then be trimmed down.

By "ConvexHull.by_polyreccone", do you mean to place the functions in ConvexHull? Currently ConvexHull depends on this file rather than the other way round.